presto-main/src/main/java/com/facebook/presto/execution/NodeScheduler.java

```diff
@@ -67,7 +67,9 @@
     public NodeScheduler(NodeManager nodeManager, NodeSchedulerConfig config, NodeTaskMap nodeTaskMap)
     {
         this.nodeManager = nodeManager;
-        this.coordinatorNodeId = nodeManager.getCurrentNode().getNodeIdentifier();
+        this.coordinatorNodeIdList = nodeManager.getCoordinators().stream()
```

Comment:
rename to `coordinatorNodeIds`
extract this code snippet to a method and reuse below.